File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {

    static Map<Integer, Integer> map = new HashMap<>();

    public int climbStairs(int n) {
         // bottom up

        map.put(1, 1);
        map.put(2, 2);

        for (int i = 3; i < n + 1; i++) {
            map.put(i, map.get(i - 1) + map.get(i - 2));
        }

        return map.get(n);
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        memo = [0] * 46

        if n == 1:
            return 1
        
        if n == 2:
            return 2

        memo[1] = 1
        memo[2] = 2

        for i in range(3, n+1):
            memo[i] = memo[i-1] + memo[i-2]
        
        return memo[n]
        